src/helpers.js

```diff
@@ -48,11 +48,16 @@ function getWindowFromNode(node) {
     throw new Error(
       `It looks like you passed a \`screen\` object. Did you do something like \`fireEvent.click(screen, ...\` when you meant to use a query, e.g. \`fireEvent.click(screen.getBy..., \`?`,
     )
-  } else {
+  } else if (node instanceof HTMLElement) {
```

Comment:
I don't understand how this would ever be the case. Particularly because `instanceof` would fail for nodes from other windows.

Reply:
Didn't think about that point.
How would you catch an instance of different windows then? And does that even a valid use-case to test for?
We can just remove this `else if` and keep the new else instead.
I wasn't sure when was the `else` part happening before, if you have an idea, please let me know :)

Reply:
> How would you catch an instance of different windows then? And does that even a valid use-case to test for?

What would we want to catch about them?

> And does that even a valid use-case to test for?

Why would it not be valid?

Reply:
The wording I used was problematic. What I meant was that before this change, we assumed that if we're getting to the `else` section, we're probably in a scenario where we have an element from a different Document. How would you assert on that?

And my thought was that this use-case can't happen. How can one create an element that has no document?

Reply:
>  we assumed that if we're getting to the else section, we're probably in a scenario where we have an element from a different Document. How would you assert on that?

I think the wording was just ambiguous. If the node is from a different document we could still get its `Window`.

The correct assumption that was never tested for is whether we have an `Element` at all. And we can't do that without duck typing. I would just adjust the error message and say at the bottom that argument doesn't look like it's an element at all. And for the earlier check for `defaultView` we should special case if `defaultView` is `null`. Because if we have a Node (`ownerDocument` property exists) but `defaultView` is `null` the window is truly not available. Though I don't know how to create a test case for `defaultView` being `null`. So that would be a scenario where we want to mock `defaultView` (but not `ownerDocument`).

Reply:
That makes since, thanks @eps1lon.
I pushed another iteration, can you please have a look to see if that's what you meant?